### PR TITLE
Refactor payload availability to request-response model

### PR DESCRIPTION
**Pull request by aschran**

### CODE REVIEW

File: narwhal/types/src/primary.rs [hunk 1]
```diff
@@ -721,7 +730,7 @@ pub struct WorkerReconfigureMessage {
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct WorkerSynchronizeMessage {
     pub digests: Vec<BatchDigest>,
-    pub target: PublicKey,
+    pub target: NetworkPublicKey,
```
Comment (bmwill) [inline narwhal/types/src/primary.rs:733]: The only thing i'm not sure of in this PR is changing the type used for `target` here.

**Review by bmwill — APPROVED** [review]

File: narwhal/types/src/primary.rs (hunk 1 above)
Comment (aschran) [reply narwhal/types/src/primary.rs:733]: Yeah I could try to put that back if the `PublicKey` is available where this is called (can't recall off-hand if it is). Originally I was thinking "why not send the key the worker has to use for networking if you have it?" before I realized that it has to translate that from primary key to worker key anyway on the receiving side

Comment (bmwill) [reply narwhal/types/src/primary.rs:733]: yeah might just be good to get an opinion from @akichidis on this one.

Comment (aschran) [reply narwhal/types/src/primary.rs]: Ok I already put it back but it'd be easy enough for me to undo the commit again :)

**Pull request merged**